src/layout/Footer.test.js

```diff
@@ -11,3 +11,53 @@ test('shows current year', () => {
   const currentYear = (new Date()).getFullYear()
   expect(footer.contains(currentYear)).toBe(true)
 })
+
+test('shows version from environment', () => {
+  // save original env var
+  const originalVer = process.env.REACT_APP_VERSION
+
+  const verString = '1.2.specialversionstring'
+  process.env.REACT_APP_VERSION = verString
+  const footer = shallow(<Footer version />)
+  expect(footer.contains(verString)).toBe(true)
+
+  // restore original env var
+  if (originalVer) {
+    process.env.REACT_APP_VERSION = originalVer
+  } else {
+    delete process.env.REACT_APP_VERSION
+  }
+})
+
+test('no crash if environment version not set', () => {
+  // save original env var
+  const originalVer = process.env.REACT_APP_VERSION
+  delete process.env.REACT_APP_VERSION
+  shallow(<Footer version />)
+
+  // restore original env var
+  if (originalVer) {
+    process.env.REACT_APP_VERSION = originalVer
+  }
+})
+
+test('shows version from prop', () => {
+  const verString = '1.2.specialversionstring'
+  const footer = shallow(<Footer version={verString} />)
+  expect(footer.contains(verString)).toBe(true)
+})
+
+test('handles boolean version prop values', () => {
+  let footer = shallow(<Footer version={false} />)
+  expect(footer.find('.footer-version')).toHaveLength(0)
+
+  footer = shallow(<Footer version={true} />)
+  expect(footer.find('.footer-version')).toHaveLength(1)
```

Comment:
Added these `expect` clauses so we check that the footer version is being rendered when it should and shouldn't.  Previously, the tests just tested if rendering worked but not the outcome.

Reply:
I think I need to be more careful about refreshing my pull req conversations -- twice now I've noticed a code update but missed the related comment.

Yes I originally wrote those tests to just confirm no crash; passing a boolean `false` as the prop value is technically allowed but not useful.  I think there's a valid case where passing `true` (the code reformatting removes the explicit boolean `true`, so it's just prop-exists-true) when there's no env var set will result in no version number appearing. I'll add an issue for that so i remember to come back around and fix it.